battery.cpp

```diff
@@ -32,3 +35,21 @@ void Battery::decrease_battery(int val) {
 int Battery::get_battery() { return battery; }
 
 int Battery::get_range() { return range; }
+
+void Battery::on(warnings warning) {
+  if (warning == LOW_BATTERY) {
+    this->ui->lowBatteryWarning->setPixmap(lowBatteryIcon);
+  }
+}
+
+void Battery::off(warnings warning) {
+  if (warning == LOW_BATTERY) {
+    this->ui->lowBatteryWarning->setText("OFF");
+  }
+}
+
+void Battery::check_range(warnings warning) {
+  if (get_range() <= 30) {
+    on(warning);
```

Comment:
Make sure to add a `off(warning)` to ensure that the warning turns off after it has been on and the range is larger than 30 miles